Migrate getTopicList() to DataProvider

BenHenning opened the pull request:
``TopicListController.getTopicList()`` was just a ``LiveData`` before which prevents us from leveraging other ``DataProviders`` tools. This PR migrates it to a ``DataProvider`` to follow the general pattern to always provide data from controllers in ``DataProvider``s and to let us interop better with other ``DataProvider``s.

BenHenning committed:
Migrate getTopicList() to DataProvider.

TopicListController.getTopicList() was just a LiveData before, and this
disallows us from leveraging other DataProviders tools. Plus, the
general pattern now is for all asynchronous data in controllers to be
given as DataProviders.

BenHenning commented:
@jcqli PTAL.

jcqli commented:
@BenHenning ok I think I'm understanding this now, thanks! FYI the robolectric tests are failing on cases that I think are related to this change

BenHenning committed:
Fix broken reference to nav drawer test.

BenHenning committed:
Add .bazelrc

This file comes from https://github.com/oppia/oppia-android/pull/1904. We need it to simplify Bazel build commands locally (bazelrc lets us automatically add arguments to certain commands like build to make it easier to type the commands in CLI locally). This file will also be used by Android Studio when using the Android Studio-with-Bazel plugin.

BenHenning committed:
Merge branch 'add-bazelrc-file' into change-get-topic-list-to-data-provider

BenHenning committed:
Fix broken tests.

These new syncs are needed since we're no longer using an immediate
LiveData for the topic list--background threads need to be sync'd
against the UI.

BenHenning commented:
Thanks @jcqli! The tests should be fixed now. PTAL!

BenHenning commented:
I think the two failures so far are flakes. Will re-run when app module tests finish.

jcqli reviewed (approved):
This looks good and all makes sense to me!

BenHenning commented:
Thanks @jcqli!
